exercises/queen-attack/src/example/java/QueenAttackCalculator.java

```diff
@@ -0,0 +1,86 @@
+import java.util.Arrays;
+
+public final class QueenAttackCalculator {
+
+    private final int[] queen1Coordinates;
+
+    private final int[] queen2Coordinates;
+
+    public QueenAttackCalculator(final int[] queen1Coordinates, final int[] queen2Coordinates)
```

Comment:
R.E. `Coordinates`: ah, thanks for the reference to the allergies exercise; symlinking into the example source set is a nice solution there. What are your thoughts, then, for the queen-attack exercise: require the implementer to build the `Coordinate` class, or provide it and allow the implementer to focus on the attack logic only?